src/pkg/utils/helpers/url.go

```diff
@@ -40,3 +41,14 @@ func DoHostnamesMatch(url1 string, url2 string) (bool, error) {
 
 	return parsedURL1.Hostname() == parsedURL2.Hostname(), nil
 }
+
+// ExtractFilenameFromURL returns filename from URL string
+func ExtractFilenameFromURL(urlStr string) (string, error) {
```

Comment:
```suggestion
func ExtractBasePathFromURL(urlStr string) (string, error) {
```

I think BasePath might be more accurate than filename since that is the function we are running under the hood.